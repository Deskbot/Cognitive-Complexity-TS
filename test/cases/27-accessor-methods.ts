var property;

class Getter {
    // depth 0
    get f(): string {
        // depth 0
        return this.f; // +1 for recursion
    }

    get [property](): string {
        // just trying to prove it doesn't fall over
        // not worth figuring out whether there's recursion here
        return ""
    }
}

class Setter {
    set f(value) {
        this.f = 1;
    }

    set [property](value) {
        // just trying to prove it doesn't fall over
        // not worth figuring out whether there's recursion here
    }
}
