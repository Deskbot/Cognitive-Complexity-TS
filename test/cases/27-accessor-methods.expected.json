{
    "27-accessor-methods.ts": {
        "score": 4,
        "inner": [
            {
                "name": "Getter",
                "score": 2,
                "line": 3,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 5,
                        "column": 5
                    },
                    {
                        "name": "[property]",
                        "score": 1,
                        "line": 11,
                        "column": 5
                    }
                ]
            },
            {
                "name": "Setter",
                "score": 2,
                "line": 17,
                "column": 1,
                "inner": [
                    {
                        "name": "f",
                        "score": 1,
                        "line": 18,
                        "column": 5
                    },
                    {
                        "name": "[property]",
                        "score": 1,
                        "line": 22,
                        "column": 5
                    }
                ]
            }
        ]
    }
}